frontend/src/metabase/collections/components/CollectionIcon.jsx

```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import Icon from "metabase/components/Icon";
+import { getCollectionIcon } from "metabase/entities/collections";
+
+const propTypes = {
+  collection: PropTypes.shape({
+    authority_level: PropTypes.oneOf(["official"]),
+  }),
+};
+
+export function CollectionIcon({ collection, ...props }) {
+  const icon = getCollectionIcon(collection);
+  return <Icon name={icon.name} color={icon.color} {...props} />;
```

Comment:
Nit

Could also be

```suggestion
  const {name, color} = getCollectionIcon(collection);
  return <Icon name={name} color={color} {...props} />;
```

Reply:
Like it! Will commit it myself as prettier would not be happy about the whitespace